python/ray/tests/test_metrics_agent.py

```diff
@@ -31,9 +31,13 @@
 # NOTE: Commented out metrics are not available in this test.
 # TODO(Clark): Find ways to trigger commented out metrics in cluster setup.
 _METRICS = [
+    # TODO(rickyx): refactoring the below 3 metric seem to be a bit involved
+    # , e.g. need product side changes on the metric subscribers.
     "ray_object_store_available_memory",
```

Comment:
Looks like we used to assume no tags on such metrics. And product has some subscribers here: https://github.com/ray-project/ray/blob/1ce0ab7b7cc6a111140e3f54f02c02725e5cc647/python/ray/autoscaler/aws/cloudwatch/example-cloudwatch-agent-config.json#L37-L39

I am not sure how product could support metrics with tags. cc @alanwguo if you know any pointers?

Reply:
Can you avoid changing old metrics and only add the new resource one? We can deprecate and remove those at a later time.